### DISCUSSION TITLE
Unable to compile on Ubuntu 24.10 / PHP 8.3 error: assignment to 'zend_string *

### DISCUSSION THREAD

**Discussion by purdytx** [phalcon/cphalcon#16657] [General]
attempting to compile latest Phalcon release via pecl results in a make failed message. Can someone help point on best path forward here? Thank you!

cphalcon/ext/kernel/main.c: In function 'zephir_function_exists_ex':
cphalcon/ext/kernel/main.c:275:83: warning: comparison between pointer and integer
  275 |         if (zend_hash_str_exists(CG(function_table), function_name, function_len) != NULL) {
      |                                                                                   ^~
cphalcon/ext/phalcon.c: In function 'php_zephir_init_globals':
cphalcon/ext/phalcon.c:1343:57: error: assignment to 'zend_string *' {aka 'struct _zend_string *'} from incompatible pointer type 'char *' [-Wincompatible-pointer-types]
 1343 |         phalcon_globals->orm.resultset_prefetch_records = ZSTR_VAL(zend_string_init(ZEND_STRL("0"), 0));
      |                                                         ^
make: *** [Makefile:210: phalcon.lo] Error 1
make: *** Waiting for unfinished jobs....
cphalcon/ext/phalcon.c: In function 'php_zephir_init_globals':
cphalcon/ext/phalcon.c:1343:57: error: assignment to 'zend_string *' {aka 'struct _zend_string *'} from incompatible pointer type 'char *' [-Wincompatible-pointer-types]
 1343 |         phalcon_globals->orm.resultset_prefetch_records = ZSTR_VAL(zend_string_init(ZEND_STRL("0"), 0));
      |                                                         ^
make: *** [Makefile:210: phalcon.lo] Error 1

**Comment by kgrammer** [comment]
How much memory do you have on your server?  PECL has issues running on small (2gb) servers.

**Comment by oleghind** [comment]
You need to set permissive flag for gcc14+, this might work:
`export CFLAGS="$CFLAGS -fpermissive"`
